# braker
Braker container generated by [Nathan Weeks](https://www.ars.usda.gov/people-locations/person?person-id=41062). Copying his instructions from the mail below:


## Instructions to run

Pull the container:

```bash
singularity pull --sandbox braker2.sif shub://aseetharam/braker
```

This will create a `braker2.sif` image, that has `braker` installed in it.


Before running, copy `augustus_config` directory as it needs to be writeable:

```bash
singularity exec braker2.sif cp -R /usr/local/config augustus_config
```

For running on your data:
```bash
readonly SINGULARITY_IMAGE=/path/to/braker2.sif
readonly BAM=/path/to/rnaseq.bam
readonly GENOME=/path/to/genome-masked.fasta
readonly PROT_SEQ=/path/to/mikado-proteins.faa
readonly SPECIES="unique-name"

# gffread adds `.` for strop codons, replace it with `*`
sed '/^[^>]/s/\./*/' ${PROT_SEQ} > ${PROT_SEQ##*/}.new

singularity exec ${SINGULARITY_IMAGE} cp -R /usr/local/config augustus_config

env time -v singularity run \
    --no-home \
    --home /opt/gm_key \
    --cleanenv \
    --env AUGUSTUS_CONFIG_PATH=${PWD}/augustus_config \
    ${SINGULARITY_IMAGE} \
        --cores ${SLURM_JOB_CPUS_PER_NODE} \
        --species=${SPECIES} \
        --genome=${GENOME} \
        --bam=${BAM} \
        --prot_seq=${PROT_SEQ} \
        --prg=gth \
        --gth2traingenes \
        --gff3
```